File: src/main/java/com/back/domain/member/member/enums/MemberStatus.java
package com.back.domain.member.member.enums;

public enum MemberStatus {
    ACTIVE,        // 정상
    SUSPENDED,     // 일시 정지
    WITHDRAWN,     // 회원 탈퇴
    BANNED;         // 영구 정지

    public boolean canTransitionTo(MemberStatus target) {
        return switch (this) {
            case ACTIVE ->
                    target == SUSPENDED || target == BANNED || target == WITHDRAWN;

            case SUSPENDED ->
                    target == ACTIVE || target == BANNED || target == WITHDRAWN;

            case BANNED ->
                    target == ACTIVE || target == WITHDRAWN;

            case WITHDRAWN -> false; // 여기서 끝
        };
    }
}

File: src/main/java/com/back/domain/member/member/enums/MemberStatus.kt
package com.back.domain.member.member.enums

enum class MemberStatus {
    ACTIVE,  // 정상
    SUSPENDED,  // 일시 정지
    WITHDRAWN,  // 회원 탈퇴
    BANNED; // 영구 정지

    fun canTransitionTo(target: MemberStatus): Boolean =
        when (this) {
            MemberStatus.ACTIVE    -> target in setOf(SUSPENDED, BANNED, WITHDRAWN)
            MemberStatus.SUSPENDED -> target in setOf(ACTIVE, BANNED, WITHDRAWN)
            MemberStatus.BANNED    -> target in setOf(ACTIVE, WITHDRAWN)
            MemberStatus.WITHDRAWN -> false
        }
}
